sourcecode/apis/contentauthor/app/Http/Controllers/H5PController.php

```diff
@@ -416,7 +416,9 @@ public function update(H5PStorageRequest $request, H5PContent $h5p, H5PCore $cor
 
         $responseValues = [
             'url' => $this->getRedirectToCoreUrl(
-                $newH5pContent->toLtiContent(),
+                $newH5pContent->toLtiContent(
+                    published: $request->validated('isPublished'),
```

Comment:
This should use the value from `$newH5pContent`, the value stored and the value in the request could be different depending on config `feature.enableUserPublish`
https://github.com/cerpus/Edlib/blob/d85df0631e8bdbe2246d46f232a93aa68f2316b2/sourcecode/apis/contentauthor/app/Libraries/H5P/h5p.php#L68

Reply:
I don't entirely agree with this. The PR only affects returns to Edlib 3, where the hub is regarded as the source of truth for publish status. It's therefore irrelevant that the value in the database doesn't match, and the long term goal is to remove the redundant flags in CA anyway.

Ideally (up for debate), if the publish toggle is disabled, the publish flag is left out entirely from the response to the hub, so it can decide whether content is published or not. However, it seems the parameter is still included with that setting. I will make a change to exclude the parameter from validation if the toggle is disabled, and keep the existing behaviour for Edlib 2.

Let me know if this sounds reasonable.